tests/tools/stagehand_tool_test.py

```diff
@@ -0,0 +1,321 @@
+def test_error_handling(mock_stagehand, mock_get_loop, stagehand_tool):
+    """Test error handling in the tool."""
+    # Setup mock
+    mock_stagehand_instance = mock_stagehand.return_value
+    mock_stagehand_instance.page = MagicMock()
+    mock_stagehand_instance.page.act.side_effect = Exception("Browser automation error")
+    mock_stagehand_instance.session_id = "test-session-id"
+
+    # Setup mock for async execution
+    mock_loop = MagicMock()
+    mock_get_loop.return_value = mock_loop
+    mock_loop.is_running.return_value = False
+    mock_loop.run_until_complete.return_value = StagehandResult(
+        success=False, data={}, error="Error using Stagehand: Browser automation error"
+    )
+
+    # Run the tool
+    result = stagehand_tool._run(
+        instruction="Click a non-existent button", command_type="act"
+    )
+
+    # Assertions
+    assert "Error:" in result
+    assert "Browser automation error" in result
+
+
+def test_initialization_parameters():
+    """Test that the StagehandTool initializes with the correct parameters."""
+    # Create tool with custom parameters
+    tool = StagehandTool(
+        api_key="custom_api_key",
+        project_id="custom_project_id",
+        model_api_key="custom_model_api_key",
+        headless=True,
+        dom_settle_timeout_ms=5000,
+        self_heal=False,
+        wait_for_captcha_solves=False,
+        verbose=3,
+    )
+
+    # Verify the tool was initialized with the correct parameters
+    assert tool.api_key == "custom_api_key"
+    assert tool.project_id == "custom_project_id"
+    assert tool.model_api_key == "custom_model_api_key"
+    assert tool.headless is True
+    assert tool.dom_settle_timeout_ms == 5000
+    assert tool.self_heal is False
+    assert tool.wait_for_captcha_solves is False
+    assert tool.verbose == 3
+
+
+def test_close_method():
+    """Test that the close method cleans up resources correctly."""
+    # Create the tool
+    tool = StagehandTool(
+        api_key="test_api_key",
+        project_id="test_project_id",
+        model_api_key="test_model_api_key",
+    )
+
+    # Set up a mock for the stagehand instance
+    mock_stagehand = MagicMock()
+
+    # Create a mock async close method that does nothing
+    async def mock_close():
+        pass
+
+    mock_stagehand.close = mock_close
+
+    # Set the mock on the tool
+    tool._stagehand = mock_stagehand
+    tool._page = MagicMock()
+
+    # Replace asyncio methods with mocks to avoid actual async execution
+    with patch("asyncio.get_event_loop") as mock_get_loop:
+        mock_loop = MagicMock()
+        mock_get_loop.return_value = mock_loop
+        mock_loop.is_running.return_value = False
+
+        # Call the close method
+        tool.close()
+
+    # Verify resources were cleaned up
+    assert tool._stagehand is None
+    assert tool._page is None
+
+
+if __name__ == "__main__":
+    pytest.main(["-xvs", "stagehand_tool_test.py"])
```

Comment:
You don't need this. The `pytest` should be able to collect and run this file automatically